### ISSUE TITLE
Debug message for host unreachable scenario

### ISSUE THREAD

**Issue by jquatier** [jquatier/eureka-js-client#38]
Currently if there is a problem with the Eureka host or port you have specified, or if the host is unreachable, you won't see any kind of error until that connection request times out. This setting is at the OS level and can be up to 2 minutes or more. The `request` module has a timeout setting, but that only applies on a read timeout, not a connection timeout. 

I think it would be helpful to output a debug message on `register()` if the call is taking too long. This would inform the user sooner that there's a problem with with either their config or their connection. Probably like 10 seconds in would be a good idea. We could also inform them to turn on request debugging for more information using `NODE_DEBUG=request`.

Anyway, basically just a setTimeout with a debug log message that we cancel on success.

**Comment by jquatier** [comment]
Available in v2.2.0
